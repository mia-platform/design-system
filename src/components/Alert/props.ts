/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactNode } from 'react'

import { IconComponent } from '../Icon/Icon.props.ts'
import { Type } from './types.ts'

export type AlertProps = {

  /**
   * The title displayed at the top of the alert.
   */
  title?: ReactNode

  /**
   * The description providing additional details about the alert.
   */
  description?: ReactNode

  /**
   * Custom children content to be rendered inside the alert.
   */
  children?: ReactNode

  /**
   * Indicates if the alert is closable.
   */
  isClosable?: boolean

  /**
   * The type of alert.
   */
  type?: Type

  /**
   * If true, renders a compact version of the alert with reduced padding.
   */
  isCompressed?: boolean

  /**
   * A custom icon to display in the alert. Replaces the default icon based on the `type`.
   */
  icon?: IconComponent

  /**
   * A custom action element (e.g., a button) to display inside the alert.
   */
  action?: ReactNode

  /**
   * A callback function triggered when the alert is closed.
   */
  onClose?: () => void
}
