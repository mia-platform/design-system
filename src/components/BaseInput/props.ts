import { InputRef } from 'antd'
import { Ref } from 'react'

import { Appearance } from './types.ts'

export type BaseInputProps = {

  /**
   * The placeholder for the input.
   */
  placeholder?: string

  /**
   * The type of the Input.
   */
  appearance?: Appearance

  /**
   * Whether the input is disabled.
   */
  isDisabled?: boolean,

  /**
   * Whether the input has error.
   */
  isError?: boolean

  /**
   * Whether the input is read-only.
   */
  isReadOnly?: boolean

  /**
   * Whether the input is full width. Defaults to true.
   */
  isFullWidth?: boolean

  /**
   * Ref to the antd input.
   */
  inputRef?: Ref<InputRef>

  /**
   * Classes for the input.
   */
  className?: string

}
