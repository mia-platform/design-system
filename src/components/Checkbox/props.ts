/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { CheckboxChangeEvent } from 'antd/es/checkbox'
import { ReactNode } from 'react'

export type CheckboxProps = {

  /**
   * Additional description of the checkbox, to be rendered below the checkbox `text`.
   * Is ignored if the `text` prop is not provided.
   */
  description?: ReactNode,

  /**
   * Allows you to control whether the checkbox is checked.
   */
  isChecked?: boolean,

  /**
   * Allows you to control whether the checkbox is disabled. Defaults to false.
   */
  isDisabled?: boolean,

  /**
   * Allows you to control whether the checkbox is checked on its first render. Defaults to false.
   */
  isInitiallyChecked?: boolean,

  /**
   * Allows you to control whether the checkbox is indeterminate. Defaults to false.
   */
  isIndeterminate?: boolean

  /**
   * Function that is invoked when the checkbox state is changed.
   */
  onChange?: (event: CheckboxChangeEvent) => void

  /**
   * Text to be displayed next to the checkbox.
   */
  text?: ReactNode
}
