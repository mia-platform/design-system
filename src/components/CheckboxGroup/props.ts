/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import { ReactNode } from 'react'

import { Direction } from './types.ts'

export type CheckboxGroupOption<T> = {

  /**
   * Adds the Radio id attribute value
   */
  value: T;

  /**
   * The text used to display the radio option
   */
  label: ReactNode;

  /**
   * description of the label that is placed underneath the label
   */
  description?: ReactNode;

  /**
   * Specifies whether the Radio option is disabled
   */
  disabled?: boolean;

}

export type CheckboxGroupProps<T> = {

  /**
   * The selected value displayed on first render
   */
  defaultValue?: T[];

  /**
   * Used for setting the currently selected value
   */
  value?: T[];

  /**
   * Options of the checkbox group
   */
  options: CheckboxGroupOption<T>[];

  /**
   * The callback function that is triggered when the state changes
   */
  onChange?: (checkedValue: T[]) => void;

  /**
   * If true, disable all checkboxes
   */
  isDisabled?: boolean;

  /**
   * The direction of the radio group. Defaults to 'vertical'
   */
  direction?: Direction
}
