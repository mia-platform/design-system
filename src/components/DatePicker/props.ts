/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  NoUndefinedRangeValueType as AntNoUndefinedRangeValueType,
  RangeValueType as AntRangeValueType,
} from 'rc-picker/lib/PickerInput/RangePicker'
import {
  RangeTimeProps as AntRangeTimeProps,
} from 'rc-picker/lib/interface'
import { Dayjs } from 'dayjs'

import { ShowTimeOptions } from './types'

type CommonProps = {
  canClear?: boolean
  format?: string
  minDate?: Dayjs
  maxDate?: Dayjs
  showTime?: boolean | ShowTimeOptions
  isErrorStatus?: boolean
  hasNowButton?: boolean
}

export type DatePickerProps = CommonProps & {
  onChange?: (date: Dayjs | Dayjs[], dateString: string | string[]) => void
  placeholder?: string
  defaultValue?: Dayjs
  value?: Dayjs
  isDisabled?: boolean
}

export type NoUndefinedRangeValueType = AntNoUndefinedRangeValueType<Dayjs>
export type RangeValueType = AntRangeValueType<Dayjs>
export type RangeTimeProps = AntRangeTimeProps<Dayjs>

export type RangePickerProps = CommonProps & {
  canBeEmpty?: boolean
  format?: string
  onChange?: (dates: NoUndefinedRangeValueType | null, dateStrings: [string, string]) => void
  placeholder?: [string, string]
  showTime?: boolean | ShowTimeOptions
  value?: RangeValueType
  defaultValue?: RangeValueType
  isDisabled?: boolean | [boolean, boolean]
}
