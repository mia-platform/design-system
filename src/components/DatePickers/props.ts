/**
 * Copyright 2025 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Dayjs } from 'dayjs'

import { NoUndefinedRangeValueType, RangeValueType, ShowTimeOptions } from './types'

type CommonProps = {

  /**
   * If false remove the clear button
   */
  allowClear?: boolean

  /**
   * To set the date format, refer to https://day.js.org/docs/en/display/format
   */
  format?: string

  /**
   * The minimum date, which also limits the range of panel switching
   */
  minDate?: Dayjs

  /**
   * The maximum date, which also limits the range of panel switching
   */
  maxDate?: Dayjs

  /**
   * To provide an additional time selection
   */
  showTime?: boolean | ShowTimeOptions

  /**
   * Error validation status
   */
  isErrorStatus?: boolean

  /**
   * Show the fast access of current datetime
   */
  showNow?: boolean
}

export type DatePickerProps = CommonProps & {

  /**
   * Callback function when the selected datetime changes
   */
  onChange?: (date: Dayjs | Dayjs[], dateString: string | string[]) => void

  /**
   * The placeholder of the date input
   */
  placeholder?: string

  /**
   * To set default date
   */
  defaultValue?: Dayjs

  /**
   * To set date
   */
  value?: Dayjs

  /**
   * To disable the datepicker
   */
  isDisabled?: boolean
}

export type RangePickerProps = CommonProps & {

  /**
   * Allow to leave start or end date empty
   */
  allowEmpty?: boolean

  /**
   * Callback function when the selected datetime changes
   */
  onChange?: (dates: NoUndefinedRangeValueType | null, dateStrings: [string, string]) => void

  /**
   * The placeholder of date input
   */
  placeholder?: [string, string]

  /**
   * To set dates
   */
  value?: RangeValueType

  /**
   * To set default dates
   */
  defaultValue?: RangeValueType

  /**
   * To disable the start or end datepicker
   */
  isDisabled?: boolean | [boolean, boolean]
}
