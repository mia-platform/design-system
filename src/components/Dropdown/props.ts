/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { KeyboardEvent, MouseEvent, ReactElement, ReactNode } from 'react'

import { Placement } from './types'
import { Size } from '../../themes/schema/shape'

export enum ItemLayout {
  Horizontal = 'horizontal',
  Vertical = 'vertical'
}

export type DropdownItem = {

  /**
   * renders the icon in a danger fashion, with red tones.
   */
  danger?: boolean,

  /**
   * Optional icon for the menu item
   */
  icon?: ReactNode

  /**
   * The id for the item.
   */
  id: string,

  /**
   * The label to be displayed for the item.
   *
   * When the automatic dropdown search is enabled via the `isSearchable` prop,
   * it will only filter items with a **string** or **number** label.
   * For labels that are ReactNodes, filtering must be handled externally via the `onSearch` callback.
   */
  label: ReactNode,

  /**
   * A secondary label to be displayed for the item.
   */
  secondaryLabel?: ReactNode,

  /**
   * A Tag, displayed on the right side of the item.
   * The argument must be a Tag component, otherwise an error will be thrown.
   */
  tag?: ReactNode

  /**
   * Whether menu item is disabled
   */
  disabled?: boolean

  /**
   * List of sub-menu items.
   */
  children?: DropdownItem[],
}

export enum DropdownTrigger {
  Click = 'click',
  Hover = 'hover',
  ContextMenu = 'contextMenu'
}

/**
 * Each FooterAction is rendered as a button with the provided configuration
 */
export type FooterAction = {

  /**
   * Optional icon
   * */
  icon?: ReactNode

  /**
   * label displayed
   * */
  label?: ReactNode

  /**
   * callback invoked on click.
   */
  onClick: () => void
}

export type DropdownHeader = {

  /**
   * Content displayed **above** the search input (if visible via `isSerchable` prop).
   * if the search input is not visible should be provided only one of `top` or `bottom` props.
   */
  top?: ReactNode

  /**
   * Content displayed **below** the search input (if visible via `isSerchable` prop).
   * if the search input is not visible should be provided only one of `top` or `bottom` props.
   */
  bottom?: ReactNode
}

export type DropdownFooter = {

  /**
   * Optional list of actions, they are rendered between the top and bottom nodes.
   */
  actions?: FooterAction[]

  /**
   * Optional node to be rendered at the bottom of the footer.
   */
  bottom?: ReactNode

  /**
   * Optional node to be rendered at the top of the footer.
   */
  top?: ReactNode
}

export type DropdownClickEvent = {

  /**
   * original dom event.
   */
  domEvent: MouseEvent<HTMLElement> | KeyboardEvent<HTMLElement>,

  /**
   * id of the clicked item.
   */
  id: string,

  /**
   * selected item data.
   */
  item?: DropdownItem,

  /**
   * array of strings representing the selected item json path.
   */
  selectedPath: string[],
}

export enum OpenChangeInfoSource {
  Trigger = 'trigger',
  Menu = 'menu'
}

export type OpenChangeInfo = {
  source: OpenChangeInfoSource
}

export type DropdownProps = {

  /**
   * Use this to focus the dropdown element.
   */
  autoFocus?: boolean

  /**
   * The ReactElement used as anchor to trigger the Dropdown visibility.
   */
  children?: ReactElement

  /**
   * Optional header configuration.
   */
  header?: DropdownHeader

  /**
   * Optional footer configuration.
   */
  footer?: DropdownFooter

  /**
   * Set to `true` to keep the selected value highlighted after the selection.
   * Useful in case you need to keep the selected option after the dropdown is closed.
   * Defaults to `true`.
   */
  persistSelection?: boolean

  /**
   * List of items to be shown as selected at first render.
   *
   * _NOTE_: this property does not work if _persistSelection_ is set to `false`
   */
  initialSelectedItems?: string[]

  /**
   * Whether the dropdown menu is disabled.
   */
  isDisabled?: boolean

  /**
   * list of items to be rendered within the Dropdown.
   */
  items: DropdownItem[]

  /**
   * Allows to control the Dropdown label layout (accepts: horizontal, vertical).
   */
  itemLayout?: ItemLayout

  /**
   * Sets the max height for the menu item list, this is useful to prevent creating
   * a dropdown with many items that may be too big (default: 240px).
   */
  menuItemsMaxHeight?: Size

  /**
   * control whether to allow multiple highlight selection.
   *
   * _NOTE_: this property does not work if _persistSelection_ is set to `false`
   */
  multiple?: boolean

  /**
   * @param DropdownClickEvent event contains the reference to the clicked item, specifically it holds
   * - id: the id of the item
   * - selectedPath: array of strings representing the selected item json path
   */
  onClick: (event: DropdownClickEvent) => void

  /**
   * list of triggers that can open the Dropdown (accepts: click, hover, contextMenu).
   */
  triggers?: DropdownTrigger[]

  /**
   * Called when the open state changes. Not triggered when hidden by click item.
   *
   * @param open the current open state of the Dropdown
   * @param info an object containing the `source` of the open change, that can be:
   * - trigger: the open state was changed by an external trigger
   *    (e.g. the dropdown is opening because the children was clicked, or closing because the focus was lost)
   * - menu: the open state was changed by clicking on a menu item
   *    (e.g. the dropdown is closing because a menu item was clicked)
   * @returns
   */
  onOpenChange?: (open: boolean, info: OpenChangeInfo) => void

  /**
   * To set the container of the dropdown menu.
   * The default behavior is to create a div element and append it at the end of the body,
   *  but you can reset it to the scrolling area and make a relative reposition.
   */
  getPopupContainer?: (triggerNode: HTMLElement) => HTMLElement

  /**
   * The placement of the dropdown menu, one of:
   *
   *  `Placement.TopLeft`, `Placement.TopRight`, `Placement.BottomLeft`,
   *  `Placement.BottomRight`, `Placement.Top`, `Placement.Bottom`
   *
   * Defaults to `Placement.BottomLeft`
   */
  placement?: Placement

  /**
   * Enables a search input for filtering items in the component.
   *
   * - If `onSearch` is **not** provided, the component applies its own
   *   built-in filtering (case-insensitive substring match on string labels).
   * - If `onSearch` **is** provided, filtering is delegated to the parent
   *   component via the callback.
   */
  isSearchable?: boolean

  /**
   * Called when the user submits a search query.
   *
   * Providing this prop overrides the component's built-in filtering logic,
   * allowing the parent component to handle search behavior.
   *
   * @param query - The text entered by the user into the search input.
   */
  onSearch?: (query: string) => void

  /**
   * Placeholder text to show inside the search input.
   * Default to 'Search...'
   */
  searchPlaceholder?: string

  /**
   * Whether the component should show a loading state.
   */
  isLoading?: boolean

  /**
   * Whether the component should show an error state.
   */
  hasError?: boolean

  /**
   * Error message to display when `hasError` is true.
   * Default to 'An error occurred'
   */
  errorMessage?: string

  /**
   * Callback fired when the user retries after an error.
   *
   * @param query - The last search query string, if available.
   */
  onRetry?: (query: string) => void

  /**
   * Enables infinite scrolling functionality for the dropdown.
   * When enabled, the `onScrollEndReached` callback will be triggered
   * when the user scrolls near the bottom of the items list.
   */
  isInfiniteScrollEnabled?: boolean

  /**
   * Callback fired when the user scrolls near the bottom of the dropdown items.
   * This is typically used to load more items for infinite scrolling.
   * Only triggered when `isInfiniteScrollEnabled` is true.
   */
  onScrollEndReached?: () => void

  /**
   * Additional items to append to the existing dropdown items.
   * These items are typically loaded asynchronously and added to the dropdown
   * when infinite scrolling loads more data.
   */
  incrementalItems?: DropdownItem[]

  /**
   * Indicates whether additional items are currently being loaded.
   * When true, a loading skeleton is displayed at the bottom of the dropdown
   * to provide visual feedback during the loading process.
   */
  isLoadingIncrementalItems?: boolean

  /**
   * Distance in pixels from the bottom of the scrollable area
   * that triggers the `onScrollEndReached` callback.
   * Defaults to 32 pixels.
   */
  scrollThreshold?: number
};
