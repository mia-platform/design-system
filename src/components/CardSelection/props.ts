/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactNode } from 'react'

import { InputType, Layout } from './types.ts'
import { IconComponent } from '../Icon/Icon.props.ts'

export type CardSelectionOption<T> = {

  /**
   * Optional id value for the card.
   */
  id?: string

  /**
   * The main title displayed prominently on the card.
   */
  title: ReactNode

  /**
   * An optional icon displayed alongside the title.
   */
  icon?: IconComponent

  /**
   * A subtitle displayed below the title.
   */
  subtitle?: ReactNode

  /**
   * Custom content to be rendered inside the card.
   */
  content?: ReactNode

  /**
   * The value associated with the card.
   */
  value?: T

  /**
   * Determines if the card is disabled.
   */
  disabled?: boolean
}

export type CardSelectionProps<T> = {

  /**
   * The default value for the card selection.
   * Must be an array if inputType is "checkbox", a value if inputType is "radio".
   */
  defaultValue?: T | T[]

  gap?: number

  /**
   * The inputType of the card selection. Defaults: undefined (no input will be displayed).
   */
  inputType?: InputType,

  isDisabled?: boolean

  /**
   * The layout of the cards. Can be "horizontal" or "vertical". Default: "vertical".
   */
  layout: Layout

  /*
  * Handler triggered when the value of the selection changes (only triggered if inputType is "checkbox" or "radio")
  * */
  onChange?: (value: T | T[]) => void

  /*
  * Handler triggered when a card is clicked
  * */
  onClick?: (value?: T) => void

  /**
   * The options array for the card selection.
   */
  options: CardSelectionOption<T>[]

  /**
   * The controlled value for the card selection.
   * Must be an array if inputType is "checkbox", a value if inputType is "radio".
   */
  value?: T | T[]
}
