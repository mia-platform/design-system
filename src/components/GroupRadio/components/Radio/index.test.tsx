import { render } from '@testing-library/react'

import { Radio } from './index'
import { RadioProps } from './index.props'

const baseProps: RadioProps = {
  label: 'label test',
  value: 'value test',
}

describe('Radio', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly with label and value props', () => {
    const { asFragment } = render(<Radio {...baseProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly with label, value and description props', () => {
    const { asFragment } = render(
      <Radio {...{ ...baseProps, description: 'description test' }} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
