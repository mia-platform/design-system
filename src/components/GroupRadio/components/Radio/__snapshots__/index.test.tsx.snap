// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly with label and value props 1`] = `
<DocumentFragment>
  <div
    class="radio"
  >
    <label
      class="ant-radio-wrapper css-dev-only-do-not-override-5wsri9"
    >
      <span
        class="ant-radio ant-wave-target"
      >
        <input
          class="ant-radio-input"
          type="radio"
          value="value test"
        />
        <span
          class="ant-radio-inner"
        />
      </span>
      <span>
        <div
          class="radioContent"
        >
          <div
            class="radioLabel"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              label test
            </div>
          </div>
        </div>
      </span>
    </label>
  </div>
</DocumentFragment>
`;

exports[`Radio renders correctly with label, value and description props 1`] = `
<DocumentFragment>
  <div
    class="radio"
  >
    <label
      class="ant-radio-wrapper css-dev-only-do-not-override-5wsri9"
    >
      <span
        class="ant-radio ant-wave-target"
      >
        <input
          class="ant-radio-input"
          type="radio"
          value="value test"
        />
        <span
          class="ant-radio-inner"
        />
      </span>
      <span>
        <div
          class="radioContent"
        >
          <div
            class="radioLabel"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              label test
            </div>
          </div>
          <div
            class="radioDescription"
          >
            <div
              class="ant-typography bodyS css-dev-only-do-not-override-5wsri9"
              role="paragraph"
            >
              description test
            </div>
          </div>
        </div>
      </span>
    </label>
  </div>
</DocumentFragment>
`;
